storage-model/src/test/java/no/unit/nva/publication/storage/model/daos/WithPrimaryKeyTest.java

```diff
@@ -0,0 +1,128 @@
+package no.unit.nva.publication.storage.model.daos;
+
+import static no.unit.nva.publication.StorageModelTestUtils.randomString;
+import static no.unit.nva.publication.storage.model.DatabaseConstants.RESOURCES_TABLE_NAME;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import com.amazonaws.services.dynamodbv2.model.PutItemRequest;
+import com.amazonaws.services.dynamodbv2.model.QueryRequest;
+import java.net.MalformedURLException;
+import java.time.Clock;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import no.unit.nva.identifiers.SortableIdentifier;
+import no.unit.nva.model.Publication;
+import no.unit.nva.model.exceptions.InvalidIssnException;
+import no.unit.nva.publication.PublicationGenerator;
+import no.unit.nva.publication.service.ResourcesDynamoDbLocalTest;
+import no.unit.nva.publication.storage.model.DoiRequest;
+import no.unit.nva.publication.storage.model.Message;
+import no.unit.nva.publication.storage.model.Resource;
+import no.unit.nva.publication.storage.model.UserInstance;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class WithPrimaryKeyTest extends ResourcesDynamoDbLocalTest {
+
+    private static final Clock CLOCK = Clock.systemDefaultZone();
+
+    @BeforeEach
+    public void init() {
+        super.init();
+    }
+
+    @ParameterizedTest
+    @MethodSource("withPrimaryKeyInstancesProvider")
+    public void primaryKeyPartitionKeyConditionReturnsQueryConditionForListingAllEntriesOfSpecificTypeForUser(
+        List<?> daos) {
+        daos.forEach(this::insertToDb);
+        WithPrimaryKey queryObject = (WithPrimaryKey) daos.get(0);
+
+        var result = performQuery(queryObject);
+        Object[] expectedItems = new Object[daos.size()];
```

Comment:
You're fond of doing it this way, but I think

    var expectedItems = daos.toArray(Object[]::new);

 is cleaner

Reply:
I hadn't thought of this way :)